SpringProject/src/main/java/com/gradproject2019/users/service/UserServiceImpl.java

```diff
@@ -15,22 +18,62 @@
 
     private final UserRepository userRepository;
 
+    public static final String PASSWORD_PATTERN = "((?=.*[a-z])(?=.*[0-9])(?=.*[!?\\#@^&£$*+;:~])(?=.*[A-Z]).{8,16})";
```

Comment:
Question: should the regexs be in this class? Maybe somewhere like AuthUtils could be better (not so sure about this though)

Reply:
Also for extra clarify something like `PASSWORD_VALIDATION_PATTERN" would be a better name, to make it clear what it's used for, and what it should be used for

Reply:
Hi. The names have been changed for password, username and email regex to include the extra clarity as suggested. Originally we had the regexs in AuthUtils but decided they might be better placed in UserServiceImpl as that is where they are being used